core/src/main/java/org/frankframework/http/MicrometerHttpClientInterceptor.java

```diff
@@ -42,12 +43,15 @@
  */
 public class MicrometerHttpClientInterceptor {
 
-	private final Map<HttpContext, Timer.ResourceSample> timerByHttpContext = new ConcurrentHashMap<>();
+	private final Map<ThreadLocal<HttpContext>, Timer.ResourceSample> timerByHttpContext = new ConcurrentHashMap<>();
 
 	private final HttpRequestInterceptor requestInterceptor;
 
 	private final HttpResponseInterceptor responseInterceptor;
 
+	// Wrap the httpContext in a threadLocal since the same context can be used over multiple threads.
```

Comment:
Shouldn't this say
```suggestion
	// Wrap the httpContext in a threadLocal since the same instance can be used over multiple threads, each of which should have their own HttpContext.
```

Reply:
Nope, het probleem is juist dat `HttpContext#1234` elke keer wordt gebruikt (met totaal andere inhoud). Maar, het is dezelfde instance waardoor het altijd vervangen wordt in de map

Reply:
Kan je de `Timer.ResourceSample` niet direct in een threadlocal zetten?

Reply:
goed idee, waarom nog in een map .. ;-) Werkt!